src/code/client.coffee

```diff
@@ -103,6 +103,38 @@ class CloudFileManagerClient
   connect: ->
     @_event 'connected', {client: @}
 
+  #
+  # Called from CloudFileManager.clientConnect to process the URL parameters
+  # and initiate opening any document specified by URL parameters. The CFM
+  # hash params are processed here after which providers are given a chance
+  # to process any provider-specific URL parameters. Calls ready() if no
+  # initial document opening occurs.
+  #
+  processUrlParams: ->
+    # process the hash params
+    hashParams = @appOptions.hashParams
+    if hashParams.sharedContentId
+      @openSharedContent hashParams.sharedContentId
+    else if hashParams.fileParams
+      if hashParams.fileParams.indexOf("http") is 0
+        @openUrlFile hashParams.fileParams
+      else
+        [providerName, providerParams] = hashParams.fileParams.split ':'
+        @openProviderFile providerName, providerParams
+    else if hashParams.copyParams
+      @openCopiedFile hashParams.copyParams
+    else
+      # give providers a chance to process url params
+      for provider in @state.availableProviders
+        handleResult = provider.canHandleUrlParams()
+        if handleResult and handleResult.providerName
+          {providerName, providerParams} = handleResult
+          @openProviderFile providerName, providerParams
+          return
```

Comment:
My thinking was that the preexisting code assumed that the result of processing the hash parameters was a call to openProviderFile; this just extended the same assumption to support resolving the query parameters before a call to openProviderFile. But I tried it out the other way and the code got simpler, so I've added a commit which refactors `canHandleUrlParams` to `handleUrlParams` with the semantics you suggest.